### PR TITLE
Nullable: AsyncLocal

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Comment by stephentoub** [comment]
Fixed.

File: src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs [hunk 1]
```diff
@@ -61,7 +62,7 @@ public T Value
             get
             {
                 object obj = ExecutionContext.GetLocalValue(this);
-                return (obj == null) ? default : (T)obj;
+                return (obj == null) ? default! : (T)obj; // TODO-NULLABLE: Need a way to annotate that Value may return null
```
Comment (safern) [inline src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs]: Should we probably add like a generic reference to these comments? So once the generic stuff lands we can grep on all this comments and fix those? Something like `// TODO-NULLABLE-GENERIC`?

File: src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs [hunk 2]
```diff
@@ -72,8 +73,8 @@ public T Value
         void IAsyncLocal.OnValueChanged(object previousValueObj, object currentValueObj, bool contextChanged)
```
Comment (safern) [inline src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs]: `previousValueObj` and `currentValueObj` are compared to null should they be nullable? If so, should we also change IAsyncLocal interface?

**Review by safern — APPROVED** [review]
Other than two questions, LGTM

File: src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs]: Yup, fixed. Thanks.

File: src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Threading/AsyncLocal.cs]: Done

**Pull request merged**